pkg/buildkit/buildkit.go

```diff
@@ -123,10 +124,16 @@ func DiscoverPlatformsFromReport(reportDir, scanner string) ([]types.PatchPlatfo
 			continue
 		}
 
+		// if arch is arm64, then variant should be v8
+		if report.Metadata.Config.Arch == "arm64" && report.Metadata.Config.Variant != "v8" {
+			report.Metadata.Config.Variant = "v8"
+		}
```

Comment:
im not sue I understand where this is coming from?

Reply:
there is a mismatch between how trivy and go-containerregistry handle arm64 variant, the former drops the v8 and the latter adds it. So we need some logic to handle this special case. I've changed it so its dropped from the reference instead of added to the report so its also consistent with the eventual tag (i.e. `...-patched-arm64`)

Reply:
got it, so in all cases we are removing the variant? could this cause a problem if another scanner doesnt drop it?

Reply:
yes, you're right this might run into trouble with other scanners. I think we can simply drop the variant from the report as well in that case, because I think `linux/arm64/v8` normalizes to `linux/arm64` (at least in containerd)